src/PositionManager.sol

```diff
@@ -167,6 +170,13 @@ contract PositionManager is
         _settle(currency, address(this), _getFullSettleAmount(currency));
     }
 
+    function _settlePair(Currency currency0, Currency currency1) internal {
+        // the locker is the payer or receiver
+        address caller = _msgSender();
+        _settle(currency0, caller, _getFullSettleAmount(currency0));
+        _settle(currency1, caller, _getFullSettleAmount(currency1));
```

Comment:
instead of reading deltas one currency at a time, lets write a helper function that uses sparse `exttload`

here's something I wrote in the initial versions of posm, but we deleted it
https://github.com/Uniswap/v4-periphery/blob/b73a2404c8c20222b795dc4eb1ce0482d775af8f/contracts/libraries/CurrencyDeltas.sol

you'll still need to settle the tokens individually
```solidity
Balance delta = poolManager.currencyDeltas(address(this), currency0, currency1); // 1 external call to PM
_settle(currency0, caller, delta.amount0());
_settle(currency1, caller, delta.amount1());
```